qemu/tests/cpu_device_hotplug.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import re
+import time
+
+from autotest.client.shared import error
+
+from virttest import utils_misc
+
+
+@error.context_aware
+def run(test, params, env):
+    """
+    Runs vCPU hotplug tests based on CPU device:
+    """
+
+    def hotplug(vm, current_cpus, total_cpus, vcpu_threads):
+        for cpu in range(current_cpus, total_cpus):
+            error.context("hot-pluging vCPU %s" % cpu, logging.info)
+            vm.hotplug_vcpu(cpu_id=cpu, plug_command=hotplug_cmd)
```

Comment:
CPU hot-plugging/hot-unplugging facility use core granularity on ppc. This means, for example, with specified `vcpu_threads=8 (smt=8)`, if you plugged 1 core then you would have 8 news vcpus. So I think the code here will out of range.